src/FluentDriver.php

```diff
@@ -79,7 +102,74 @@ public function isTransient($className)
     }
 
     /**
+     * @param array $paths
+     *
+     * @throws MappingException
+     */
+    public function loadPaths(array $paths)
+    {
+        $classes = [];
+        $includedFiles = [];
+
+        foreach ($paths as $path) {
+            if (!is_dir($path)) {
+                throw MappingException::fileMappingDriversRequireConfiguredDirectoryPath($path);
+            }
+
+            $iterator = new RegexIterator(
+                new RecursiveIteratorIterator(
+                    new RecursiveDirectoryIterator($path, FilesystemIterator::SKIP_DOTS),
+                    RecursiveIteratorIterator::LEAVES_ONLY
+                ),
+                '/^.+'.preg_quote($this->fileExtension).'$/i',
+                RecursiveRegexIterator::GET_MATCH
+            );
+
+            foreach ($iterator as $file) {
+                $sourceFile = $file[0];
+
+                if (!preg_match('(^phar:)i', $sourceFile)) {
+                    $sourceFile = realpath($sourceFile);
+                }
+
+                require_once $sourceFile;
+
+                $includedFiles[] = $sourceFile;
+            }
+
+            $declared = get_declared_classes();
+
+            foreach ($declared as $className) {
```

Comment:
@patrickbrouwers this list could be huge. We could use some convention to skip building a ReflectionClass for each loaded class in the system... Or maybe parse filename and class name and only reflect on basename matches.